### PR TITLE
Proof of concept springdoc-openapi

### PR DESCRIPTION
**Pull request by varunjayaraman**
Swagger UI available at `localhost:8080/docs-ui.html` and JSON docs available at `localhost:8080/docs` and you can export a YAML file with `localhost:8080/docs.yaml`

### CODE REVIEW

**Comment by 0teh** [comment]
Looks pretty good so far, I am wondering where the error responses are though? Is it possible with our current approach, or would we have to annotate everything?

**Comment by varunjayaraman** [comment]
can you clarify what you mean by that?

**Comment by 0teh** [comment]
I only see 200 responses in the UI, can we easily display error responses as well (using `ApiReponse` sealed classes)?

File: build.gradle.kts [hunk 1]
```diff
@@ -89,5 +90,5 @@ tasks {
 }
 
 tasks.withType<org.springframework.boot.gradle.tasks.bundling.BootJar> {
-    setProperty("archiveFileName", "lab-res.jar")
+    setProperty("archiveFilename", "lab-res.jar")
```
Comment (0teh) [inline build.gradle.kts:93]: This change fails the build

Comment (varunjayaraman) [reply build.gradle.kts]: yeah it was a mistake

File: src/main/kotlin/com/healthmetrix/labres/GlobalErrorHandler.kt [hunk 2]
```diff
@@ -1,20 +1,30 @@
...
         val id = UUID.randomUUID()
         logger.error("Exception caught incidentId=$id", exception)
         return InternalServerError("$id").asEntity()
     }
 
-    data class InternalServerError(val message: String) : ApiResponse(HttpStatus.INTERNAL_SERVER_ERROR)
+    @Schema(description = "General error response body")
+    data class InternalServerError(
+        @Schema(
+            description = "A summary of the error that occurred",
```
Comment (0teh) [inline src/main/kotlin/com/healthmetrix/labres/GlobalErrorHandler.kt:25]: Right now, the content of the message is actually an incident id (UUID), should probably be mentioned here

File: src/main/kotlin/com/healthmetrix/labres/order/OrderController.kt [hunk 3]
```diff
@@ -68,14 +112,24 @@ data class CreateOrderRequestBody(val notificationId: String)
...
+    LabResApiResponse(httpStatus, hasBody) {
+
     data class Created(
+        @Schema(
+            description = "A unique internal identifier for the order",
+            example = "65e524cb-7494-4073-ad16-495fed0d79e4"
+        )
         val id: UUID,
+        @Schema(
+            description = "numeric 10-digit-long identifier",
```
Comment (0teh) [inline src/main/kotlin/com/healthmetrix/labres/order/OrderController.kt:130]: EONs are mentioned above, should we reuse that vocab here or remove EON vocab from above?

**Pull request merged**